fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -457,7 +458,9 @@ public void close() {
      */
     @Nonnull
     public CompletableFuture<Void> buildRange(@Nullable Key.Evaluated start, @Nullable Key.Evaluated end) {
-        return recordStoreBuilder.getSubspaceProvider().getSubspaceAsync().thenCompose(subspace -> buildRange(subspace, start, end));
+        return runner.runAsync(context -> {
+            return recordStoreBuilder.getSubspaceProvider().getSubspaceAsync(context);
+        }).thenCompose(subspace -> buildRange(subspace, start, end));
```

Comment:
I think you actually want to modify the overload of `buildRange` to take a `SubspaceProvider` instead of a subspace and let it get resolved there. That also makes it somewhat more natural to use the `runner` to grab a context and from that a transaction that can then be passed to the `RangeSet` when getting all of the ranges instead of using the database. This has the added benefit of using the runner's retry logic (instead of the default behavior) which I think we want as well.

Reply:
Are you suggesting that missingRanges be refactored to take an FDBRecordContext? If so, there are a few other uses of this method and it might end up expanding the scope of this particular PR to do that work. Please advice.

Reply:
I don't think `missingRanges` needs to be changed. It already takes a `ReadTransactionContext` which is essentially an algebraic data type (ignoring for the moment that those don't exist in Java) that can either be a transaction or a database. We already use it a few different places in the `OnlineIndexer` in instances where it takes a transaction (those invocations that take `recordStore.ensureContextActive()` essentially extract the FDB transaction from the record store and hand it to the range set); I think we only give it a database in `buildRange`.

The difference between calling it with a transaction and calling it with a database is that if you give it a database, it will use the database to create a transaction, run the code with the transaction it just created, and then retry retriable failures, whereas if you give it a transaction, it runs the code exactly once and bubbles up failures (retriable or not) to the user. The problem is that I think we actually want to use the retry logic in the `runner` rather than the default `FDB` behavior (which is what `buildRanges` currently does).